code/modules/client/client procs.dm

```diff
@@ -109,135 +109,53 @@
 			if("shop")
 				if(href_list["KarmaBuy"])
 					var/karma=verify_karma()
-					switch(href_list["KarmaBuy"])
-						if("1")
-							if(karma <5)
-								to_chat(usr, "You do not have enough karma!")
+					if(!isnull(karma)) //Doesn't display anything if karma database is down.
+						switch(href_list["KarmaBuy"])
+							if("1")
+								karma_purchase(karma,5,"job","Barber")
 								return
-							else
-								if(alert("Are you sure you want to unlock Barber?", "Confirmation", "No", "Yes") != "Yes")
-									return
-								DB_job_unlock("Barber",5)
+							if("2")
+								karma_purchase(karma,5,"job","Brig Physician")
 								return
```

Comment:
All those returns are not needed. Switch only execute one statement if a condition is met.